gunicorn/workers/sync.py

```diff
@@ -126,8 +126,37 @@ def handle(self, listener, client, addr):
         req = None
         try:
             if self.cfg.is_ssl:
-                client = ssl.wrap_socket(client, server_side=True,
-                    **self.cfg.ssl_options)
+                # TODO: Make this reusable by gthread worker. Move to BaseWorker?
+                context = ssl.SSLContext(self.cfg.ssl_options['ssl_version'])
+                context.set_default_verify_paths()
+                ciphers = self.cfg.ssl_options['ciphers']
+                # set_ciphers() will raise an exception if we pass None (default value)
+                if ciphers is not None and isinstance(ciphers, str):
+                    try:
+                        context.set_ciphers(self.cfg.ssl_options['ciphers'])
+                    except ssl.SSLError as exc:
+                        # TODO: Log this in a better way.
+                        print(exc)
+                        raise exc
+                # TODO: OP_CIPHER_SERVER_PREFERENCE is set by default since Python 3.6+.
+                context.options |= ssl.OP_CIPHER_SERVER_PREFERENCE
```

Comment:
Any chance this can be exposed as a global configuration option? It would be very nice, particularly if you want to exclude specific protocol versions. 
e.g.
```
ssl_options = ssl.OP_NO_TLSv1 | ssl.OP_NO_TLSv1_1 | ssl.OP_NO_TLSv1_2
```

Reply:
Yes, that's definitely possible!

BTW, I think `SSLContext` in Python 3.6+ sets all of those options by default.